crates/evm/coverage/src/inspector.rs

```diff
@@ -89,14 +89,10 @@ impl LineCoverageCollector {
 /// tx) then the hash is calculated from the bytecode.
 #[inline]
 fn get_or_insert_contract_hash(interpreter: &mut Interpreter) -> B256 {
-    if interpreter.bytecode.hash().is_none_or(|h| h.is_zero()) {
-        interpreter.bytecode.regenerate_hash();
-    }
-    interpreter.bytecode.hash().unwrap_or_else(|| eof_panic())
-}
-
-#[cold]
-#[inline(never)]
-fn eof_panic() -> ! {
-    panic!("coverage does not support EOF");
+    // TODO: use just `get_or_calculate_hash`
```

Comment:
We're wasting some time by checking both `None` and `is_zero`

https://github.com/bluealloy/revm/pull/2826